primitives/src/transaction.rs

```diff
@@ -291,6 +304,58 @@ fn hash_transaction(tx: &Transaction, uses_segwit_serialization: bool) -> sha256
     sha256d::Hash::from_engine(enc)
 }
 
+/// The encoder for the [`Transaction`] type.
+#[cfg(feature = "alloc")]
+pub struct TransactionEncoder<'e>(
```

Comment:
nit: I believe you can use the `encoding::encoder_newtype!` here now if you want.

Reply:
Done, nice.